simpleviewinc/keg-hub
# keg-components: Item header update

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-261)

## Context

* The plan moving forward is to create 2 different headers
    * **ItemHeader** - can be used in other areas of the app
    * **AppHeader** - should only be used for the main App Header / native navigation header. utilizing the native apps status bar customization
* since we're using this component in the `tap-events-force`  several times already, I figured we start the separation now to avoid having to go back later

## Goal

* Convert the current `AppHeader` to be `BaseHeader` component and use that to create the new `AppHeader` and `ItemHeader`
* for consumer to start using `ItemHeader` as opposed to `AppHeader`
* separate current **AppHeader** component into 2 types. 
* **ItemHeader** will behave exactly the same as the old **AppHeader**
* **AppHeader** will eventually be updated to have the native status bar, but right now just keep it the same
* utilize `ItemHeader` in `tap-events-force` for headers that is not used at the top App level
* the way you consume `ItemHeader` is still the same as the old `AppHeader`

## Updates

* `.eslintrc.js`
    - removed this since it was causing issues in the `/repos` that has eslint

* `repos/keg-components/src/components/header/*`
   - created `appHeader`, `baseHeader` and `itemHeader` components

* `repos/keg-components/src/components/header/appHeader.js`
   - behaves exactly like `ItemHeader` for now.
   - will update with native functionality on another ticket

* `repos/keg-components/src/components/header/baseHeader.js`
   - basically the old `AppHeader`. renamed and slightly updated to be reusable for AppHeader and ItemHeader

* `repos/keg-components/src/components/header/itemHeader.js`
    - uses the BaseHeader

## Testing

* Pull this branch
* cd into keg-components
    * `keg components`
* run `yarn sb`
* verify that there is a `Header` section
* verify that `ItemHeader` looks and functions the same as the old `AppHeader`

### Custom dataSet
* go to `ItemHeader` > `plain` storybook demo
* inspect the top most header component
* verify that the `left` and `center` section has a custom dataSet
     * ![image](https://user-images.githubusercontent.com/3317835/91499706-ce1ccb80-e876-11ea-91bd-d808d8a419ef.png)


* the `right` section does not have a custom dataSet so it should default to `item-header-content-right-main`
